debezium-ai/debezium-ai-embeddings-base/src/main/java/io/debezium/ai/embeddings/AbstractEmbeddingsSmt.java

```diff
@@ -0,0 +1,171 @@
+ */
+package io.debezium.ai.embeddings;
+
+import static java.lang.String.format;
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.components.Versioned;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.transforms.ConnectRecordUtil;
+import io.debezium.transforms.SmtManager;
+import io.debezium.util.BoundedConcurrentHashMap;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+
+/**
+ * Base class of the embeddings SMT, which provides common configuration and manipulations with the Connect record.
+ * Child classes has to implement at least {@link AbstractEmbeddingsSmt#getModel()} class, which provides specific
+ * model which will be used for creating the embeddings.
+ *
+ * @author vjuranek
+ */
+public abstract class AbstractEmbeddingsSmt<R extends ConnectRecord<R>> implements Transformation<R>, Versioned {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractEmbeddingsSmt.class);
+
+    private static final Field TEXT_FIELD = Field.create("field.source")
+            .withDisplayName("Name of the record field from which embeddings should be created.")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.SHORT)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .required()
+            .withDescription("Name of the field of the record which content will be used as an input for embeddings. Supports also nested fields.");
+
+    private static final Field EMBEDDGINS_FIELD = Field.create("field.embeddings")
+            .withDisplayName("Name of the field which would contain the embeddings of the input field.")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.SHORT)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .required()
+            .withDescription(
+                    "Name of the field which which will be appended to the record and which would contain the embeddings of the content `filed.source` field. Supports also nested fields.");
+
+    private static final Field.Set ALL_FIELDS = Field.setOf(TEXT_FIELD, EMBEDDGINS_FIELD);
+    private static final Schema VECTOR_SCHEMA = SchemaBuilder.array(Schema.FLOAT32_SCHEMA).schema();
```

Comment:
I wonder, if I should use `io.debezium.data.vector.FloatVector` (IMHO yes, will include it in final version).